odl/tomo/geometry/geometry.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Geometry base class."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from abc import ABCMeta, abstractmethod, abstractproperty
+from future import standard_library
+from future.utils import with_metaclass
+standard_library.install_aliases()
+from builtins import object
+
+# External
+import numpy as np
+
+# Internal
+from odl.set.domain import IntervalProd
+from odl.discr.grid import RegularGrid, TensorGrid
+
+
+__all__ = ('Geometry',)
+
+
+class Geometry(with_metaclass(ABCMeta, object)):
```

Comment:
What about SPECT, PET maybe others in the long run (MRI?). I think this needs a more specific name.